ai-service/src/ai_service/project_ingestor.py

```
@@ -0,0 +1,93 @@
+    """
+    Clones a GitHub repo to a temporary directory.
+    Returns the path to the cloned directory.
+    Validates the repo_url to prevent command injection.
+    """
+    # Only allow URLs matching the GitHub repo pattern
+    github_repo_pattern = r"^https://github\.com/[\w\-]+/[\w\-]+(\.git)?$"
+    if not re.match(github_repo_pattern, repo_url):
+        raise errors.InvalidParam.invalid_repo_url()
+    clone_to = tempfile.mkdtemp()
+    try:
+        subprocess.run(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)